src/preamble.js

```diff
@@ -322,7 +322,7 @@ var HEAP,
   HEAPF64;
 
 function updateGlobalBuffer(buf) {
```

Comment:
I think you can remove this function completely.

It has exactly one call site and the callsite which always used the return value of `emscripten_realloc_buffer()`, but `emscripten_realloc_buffer()` already takes care up updating the global `buffer`.

Reply:
Thanks, done.